### PR TITLE
Use POST for all requests instead of GET

### LINKED ISSUE #24: Invalid secret
Hi, I'm trying to install this library.

I followed steps, so I made `composer require graze/gigya-client`, and my script:

``` php
require_once 'vendor/autoload.php';

use Graze\Gigya\Response\ErrorCode;
use Graze\Gigya\Gigya;

$key = '[my key]';
$secret = '[my secret]';
$uid = '[an existing uid]';

$gigya = new Gigya($key, $secret);

$response = $gigya->accounts()->getAccountInfo(['uid' => $uid]);
if ($response->getErrorCode() === ErrorCode::OK) {
    $account = $response->getData();
    var_dump($account);
} else {
    var_dump('FAILED', $response);
}
```

I expected to get the user account info that match the uid.

Instead, when I run it, I get an error like:

``` js
{
    "statusReason": "Forbidden",
    "errorMessage": "Invalid Secret"
}
```

But I know that my secret is valid because it works in another project.

So I debug the sent request by guzzle, and I get this host and parameters:

```
string(54) "https://accounts.eu1.gigya.com/accounts.getAccountInfo"
array(3) {
  ["verify"]=>
  string(66) "/.../vendor/graze/gigya-client/src/cacert.pem"
  ["auth"]=>
  string(5) "gigya"
  ["query"]=>
  array(1) {
    ["uid"]=>
    string(36) "[an existing uid]"
  }
}
```

I can't find where the api key and the secret is sent. In my other project, I send it here, in the request parameters.

Also it works with curl and the same parameters:

```
curl -X POST \
    https://accounts.eu1.gigya.com/accounts.getAccountInfo \
    -H 'content-type: application/x-www-form-urlencoded' \
    --data-urlencode "apiKey=[my api key]" \
    --data-urlencode "secret=[secret]" \
    --data-urlencode "UID=[an existing uid]"
```

### LINKED ISSUE #25: Move all requests to POST
there is no need for everything to be get requests.

also post will make sure the key/secret does not get exposed.

### PR DESCRIPTION
**Pull request by h-bragg**
This changes all requests to use GET.

Looks like I made a branch for this a while ago too: https://github.com/graze/gigya-client/tree/post

Based on #26 (thanks @alcalyn!)

Resolves #24 
Resolves #25 

@alcalyn are you ok to look over this and test it with your Gigya?

### CODE REVIEW

**Comment by h-bragg** [comment]
This also includes some general tidy up of the repo from our other open source repos. Which is why the tests are failing. Need to up the minimum versions.

File: tests/src/TestCase.php [hunk 1]
```diff
@@ -17,6 +17,13 @@
 
 class TestCase extends \PHPUnit_Framework_TestCase
 {
+    public function setUp()
+    {
+        if ('' === ini_get('date.timezone')) {
+            date_default_timezone_set('UTC');
```
Comment (biggianteye) [inline tests/src/TestCase.php]: For tests, wouldn't it be better to enforce a specific timezone for consistency?

Comment (h-bragg) [reply tests/src/TestCase.php]: done

**Comment by alcalyn** [comment]
I will test this version today

**Comment by alcalyn** [comment]
It works. I can make:

``` php
$response = $this->gigya->accounts()->search([
    'query' => 'SELECT count(*) FROM accounts',
]);

return $response->getData()->first()->{'count(*)'}
```

**Pull request merged**